# Problem Statement:
  Install timezone data while building glibc v2.20.

# Owned by:
  Rohit

# Actions:
  * Created a patch which updates the timezone/Makefile to build cross-zic.
  * Updated build.sh script to build & install tzdata after glibc build is
    complete.

diff -Naur glibc-2.20/timezone/Makefile glibc-2.20-tzdata/timezone/Makefile
--- glibc-2.20/timezone/Makefile	2014-09-07 03:09:09.000000000 -0500
+++ glibc-2.20-tzdata/timezone/Makefile	2015-02-18 08:02:49.041739203 -0600
@@ -25,6 +25,9 @@
 extra-objs := scheck.o ialloc.o
 
 others	:= zdump zic
+ifneq ($(cross-compiling),no)
+others	+= cross-zic
+endif
 tests	:= test-tz tst-timezone
 
 # pacificnew doesn't compile; if it is to be used, it should be included in
@@ -53,10 +56,21 @@
 
 include ../Rules
 
+zic-objs = zic.o ialloc.o scheck.o
+
+$(addprefix $(objpfx)cross-,$(zic-objs)): $(objpfx)cross-%.o: %.c
+	$(BUILD_CC) $< -c $(OUTPUT_OPTION) $(CFLAGS-$*.c) $(CPPFLAGS-$*) \
+		-DREPORT_BUGS_TO='"$(REPORT_BUGS_TO)"' \
+		-DPKGVERSION='"$(PKGVERSION)"' \
+		$(compile-mkdep-flags) -iquote $(objpfx)
+
+$(objpfx)cross-zic: $(addprefix $(objpfx)cross-,$(zic-objs))
+	$(BUILD_CC) $(addprefix $(objpfx)cross-,$(zic-objs)) -o $@
 
 $(objpfx)zic: $(objpfx)scheck.o $(objpfx)ialloc.o
 
 $(objpfx)zic.o $(objpfx)zdump.o: $(objpfx)version.h
+$(objpfx)cross-zic.o $(objpfx)cross-zdump.o: $(objpfx)version.h
 
 $(objpfx)version.h: $(common-objpfx)config.make
 	echo 'static char const TZVERSION[]="$(version)";' \
@@ -77,10 +91,17 @@
 # We have to make sure the data for testing the tz functions is available.
 # Don't add leapseconds here since test-tz made checks that work only without
 # leapseconds.
+ifeq (no,$(cross-compiling))
 define build-testdata
 $(built-program-cmd) -d $(testdata) -y ./yearistype $<; \
 $(evaluate-test)
 endef
+else
+define build-testdata
+LANGUAGE=C LC_ALL=C \
+  $(objpfx)cross-zic -d $(testdata) -y ./yearistype $<
+endef
+endif
 
 $(objpfx)test-tz.out: $(addprefix $(testdata)/, America/New_York Etc/UTC UTC)
 $(objpfx)tst-timezone.out: $(addprefix $(testdata)/, \
@@ -93,7 +114,11 @@
 tst-timezone-ENV = TZDIR=$(testdata)
 
 # Note this must come second in the deps list for $(built-program-cmd) to work.
+ifeq (no,$(cross-compiling))
 zic-deps = $(objpfx)zic $(leapseconds) yearistype
+else
+zic-deps = $(objpfx)cross-zic $(objpfx)zic $(leapseconds) yearistype
+endif
 
 $(testdata)/America/New_York: northamerica $(zic-deps)
 	$(build-testdata)
